h2o-automl/src/main/java/ai/h2o/automl/ModelingStep.java

```diff
@@ -352,4 +354,52 @@ protected Work getAllocatedWork() {
             );
         }
     }
+
+    /**
+     * Convenient class for testing time-based constrains
+    */
+    public static abstract class DummyStep<M extends Model> extends ModelingStep<M> {
+
+        public DummyStep(Algo algo, String id, int weight, AutoML autoML) {
+            super(algo, id, weight, autoML);
+        }
+
+        @Override
+        protected Work makeWork() {
+            return new Work(_id, _algo, WorkAllocations.JobType.Unknown, _weight);
+        }
+
+        @Override
+        protected Work getAllocatedWork() {
+            return getWorkAllocations().getAllocation(_id, _algo);
+        }
+
+        @Override
+        protected Key makeKey(String name, boolean withCounter) {
+            return aml().dummyKey(name, withCounter);
+        }
+
+        @Override
+        protected Job<M> startJob() {
+            // Sleep weight/10 seconds
+            long chunks = _weight;
+            Job<M> job = new Job(Key.make("dummy_model_step_sleep_job"), Model.class.getName(), "does nothing, just sleeps");
+            return job.start(new H2O.H2OCountedCompleter() {
+                @Override
+                public void compute2() {
+                    try {
+                        for (int i = 0; i < chunks; i++) {
+                            // Pretend some work
+                            Thread.sleep(100);
+                            // Update the status
+                            job.update(1);
+                        }
```

Comment:
ideally, you should also be able to have this dummy step providing a proper (but untrained) model, just use the constructor (e.g. DRFModel) and put it in DKV.
Also the idea is to use this to test automl logic for `max_runtime_secs`, making the assumption that the individual models will respect the `max_runtime_secs` param that we set for each of them.
What it means, is that the dummy step should sleep for the `max_runtime_secs` that is computed for it, by the logic in `ModelingStep`.
However, I'm afraid that the current logic there doesn't make it easy: I had to change this in my current branch though, so it will be easier once merged.

Reply:
@sebhrusen I am struggling now with creating the DRFModel without training it I always get NullPointerException (see below). If I understand the nature of the problem, AutoML tries to get metrics from the untrained model. Any ideas how to deal with that?

```
java.lang.NullPointerException
        at ai.h2o.automl.leaderboard.Leaderboard.getMetrics(Leaderboard.java:461)
        at ai.h2o.automl.leaderboard.Leaderboard.addModels(Leaderboard.java:374)
        at ai.h2o.automl.leaderboard.Leaderboard.addModel(Leaderboard.java:395)
        at ai.h2o.automl.ModelingStepsExecutor.addModel(ModelingStepsExecutor.java:170)
        at ai.h2o.automl.ModelingStepsExecutor.monitor(ModelingStepsExecutor.java:147)
        at ai.h2o.automl.ModelingStepsExecutor.submit(ModelingStepsExecutor.java:66)
        at ai.h2o.automl.AutoML.learn(AutoML.java:574)
        at ai.h2o.automl.AutoML.run(AutoML.java:377)
        at ai.h2o.automl.H2OJob$1.compute2(H2OJob.java:33)
        at water.H2O$H2OCountedCompleter.compute(H2O.java:1470)
        at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
        at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
        at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
        at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
        at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)
```

Reply:
@sebhrusen for now, I have implemented custom logic for the `max_runtime_secs`, mainly so I can test the implementation somehow.

It works like this: 
1. if `max_runtime_secs_per_model` is positive, use `min(max_runtime_secs_per_model, remaining_time)`
2. if not (1.) and `max_runtime_secs` is positive, use `min(max_runtime_secs, remaining_time)`
3. if not (1.) and not (2.), use `min(_weight, remaining_time)`

where `remaining_time` is `aml().timeRemainingMs()/1e3` and `max_runtime_secs{,_per_model}` are `aml().getBuildSpec().build_control.stopping_criteria.max_runtime_secs{,_per_model}()`.